### ISSUE TITLE
Deepsea model performance 

### ISSUE THREAD

**Issue by gouthamatla** [FunctionLab/selene#99]
Hi,

I ran selene_cli.py (as in case-I of the manuscript) on two ChIP-Seq data sets (peaks) generated from tissue of my interest. CTCF and PDX1 peaks. 

In case of CTCF, the precision and auc looks really good.  

```
cat selene_sdk.train_model.validation.txt
loss	average_precision	roc_auc
0.07284873615391553	0.6406582574587418	0.9707755515007088
0.06534399052709342	0.7066890703048329	0.9757932102045571
0.061813083751127124	0.7349686310186131	0.9790489986087729
0.06546932762907819	0.7395680971047822	0.9794652750330148
0.05991097574308515	0.7437388469286021	0.9798042120174655
0.0617053345553577	0.7500170838638921	0.9799612989750899
0.060426393245114014	0.7434227016711831	0.9807914036108969
0.062121331565547735	0.752554596009796	0.9799384623683222
0.05800173483183607	0.7489864601111719	0.9807203653086499
0.05784373829467222	0.7559693300334487	0.981081483367512
0.06287746413517743	0.7566987552674547	0.9801430954890704
0.05799780744779855	0.761240593477695	0.9825303436562824
0.0627876044602599	0.7564390843944009	0.980710358406563
```

In case of PDX1, the precision is very very low and auc is also low.
```
cat selene_sdk.train_model.validation.txt
loss	average_precision	roc_auc
0.06329237390868366	0.02749179724910813	0.6468820033919309
0.060851018268615005	0.03451037399439737	0.7321411125414345
0.06801522767543793	0.04167975643979051	0.7846509659319911
0.05850029695406556	0.05502252589299553	0.8069343991543023
0.06070629969891161	0.058933500947012524	0.8063612946300388
0.05507130430173129	0.07047871049562453	0.8268254811445133
0.054641662795096635	0.07391492950309346	0.833448224700689
0.05736865572072566	0.06446364541993471	0.8042514801198469
0.05808867362141609	0.07631723291609437	0.809285849832374
0.056033607934601604	0.0769540901134627	0.820345475554219
0.05697573825716972	0.08530214805003257	0.8167832360054698
0.05782379073044285	0.08117566918765817	0.8245975600749655
0.060361943366937336	0.09178671478487724	0.8256518670688942
```

Presumably, CTCF has a strong motif (~15bp) that the model learns, but PDX1 also has a very strong motif (4-6bp) but smaller. CTCF might not have any co-binding partners, so its a single strong motif across most of the peaks. While PDX1 might have several co-binding partners and indirect binding, there may not be a single particular motif across all peaks, and model might not be picking up the patterns.

I used `deepsea.py` model and `deepsea_TF_intervals.txt` on hg19 genome with default parameters in `case1` .yml file.

I am wondering why there is such difference.  So I have few questions:

1. Is it because the CTCF peaks are more consistent across tissues and cell types, is there anything that biases model towards better performance on CTCF ( Does  `deepsea_TF_intervals.txt` has something to do with intervals that are shared across tissues ? )

2. What are the possible parameters that I potentially could change and play around ?  **It would be nice to have a section of Critical Parameters in the tutorial.**  Should I test deeper-deepsea model ?

3. Regarding the input peaks file, should I split the peaks into smaller chunks before running selene_cli ? The parameters in yml file  ` sequence_length: 1000, center_bin_to_predict: 200` does this mean the model always takes the center 200bp with +/- 500bp for each peak ?

Thanks,
Goutham A

**Comment by kathyxchen** [comment]
We've found that CTCF is usually the easiest feature to predict in these kinds of deep learning models.

I don't really have recommendations for parameters to test - remember that Selene is a tool for model development, so there's a lot of ways to change the architecture and it's hard for me to make a specific recommendation in that respect. Deeper architectures, like we showed in the paper, do seem to have some improvement in terms of AUC. 

You can compare the differences in peak sizes between your CTCF and PDX1 tracks, look at how the measurements for PDX1 are distributed (maybe the sampler isn't labeling many sequences with PDX1), etc. You can adjust your model convolutional layers' # of channels, or bin/feature threshold/sequence length based on that.

**Comment by jzthree** [comment]
You may also need to adjust the learning rate and how long the model is trained for with different training data and models.

**Comment by gouthamatla** [comment]
Thanks, I will try with different options.

**Comment by kathyxchen** [comment]
@gouthamatla you can post follow-ups to this issue in our newly created google group https://groups.google.com/forum/#!forum/selene-sdk

Thanks!
